lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -244,6 +244,283 @@ class ConvertAtenReflectionPad1dOp
 };
 }
 
+namespace {
+
+// Lower the aten.reflection.pad_2d operator into a sequence of
+// tensor.extract_slice, linalg.generic, and tensor_insert_slice
+// operations.
+
+// To understand the lowering, consider this pytorch example:
+//
+// >>> t = torch.tensor([[[1.0,2,3],[4,5,6], [7,8,9]]])
+// >>> t
+// tensor([[[1., 2., 3.],
+//         [4., 5., 6.],
+//         [7., 8., 9.]]])
+// >>> torch.ops.aten.reflection_pad2d(t, [1,2,1,2])
+// tensor([[[5., 4., 5., 6., 5., 4.],
+//          [2., 1., 2., 3., 2., 1.],
+//          [5., 4., 5., 6., 5., 4.],
+//          [8., 7., 8., 9., 8., 7.],
+//          [5., 4., 5., 6., 5., 4.],
+//          [2., 1., 2., 3., 2., 1.]]])
+//
+// The result can be subdivided into "tiles" corresponding to either
+// the input tensor (in the center) or slices of the input tensor
+// whose width and height is determined by the padding sizes and which
+// are reflected through the side of the central input tensor that
+// they touch.
+// In the example above, the tiles are:
+// top left: [[5]]
+// top center: [[4,5,6]]
+// top right: [[5,4]]
+// center left [[2,1],[5,4],[8,7]]
+// center: copy of the input tensor
+// center right: [[2,1],[5,4],[8,7]]
+// bottom left: [[5,4],[2,1]]
+// center bottom: [[2,3,2]]
+// center right: [[2,1]]
+//
+// The lowering uses a tensor.extract_slice operation to create each tile,
+// a linalg.generic for the reflection, and a tensor.insert_slice to
+// insert the tile in the resulting tensor.
```

Comment:
Ok, thanks for the pointer. I have noticed the operation but it did not seem fitting to me. Looking at it now, I think it could be possible to use it, but I am not really sure about the details. Perhaps this should be reconsidered in the future.